test/parallel/test-runner-mocking.js

```diff
@@ -1054,3 +1054,68 @@ test('setter() fails if getter options is true', (t) => {
     t.mock.setter({}, 'method', { getter: true });
   }, /The property 'options\.setter' cannot be used with 'options\.getter'/);
 });
+
+test('spies on a property value', (t) => {
```

Comment:
I think we need more tests than this. For example, we need to ensure that mocks created with this new API on the `TestContext` are automatically restored at the end of the test.